walletlibrary/src/main/java/com/microsoft/walletlibrary/mappings/presentation/PresentationResponseMapping.kt

```diff
@@ -31,21 +31,42 @@ private fun PresentationResponse.addVerifiedIdRequirement(verifiedIdRequirement:
             "Verified ID has not been set.",
             VerifiedIdExceptions.REQUIREMENT_NOT_MET_EXCEPTION.value
         )
-    val credentialPresentationInputDescriptor =
-        request.getPresentationDefinition().credentialPresentationInputDescriptors.filter { it.id == verifiedIdRequirement.id }
-    if (credentialPresentationInputDescriptor.isEmpty())
+    val matchingInputDescriptors =
+        request.getPresentationDefinitions().filter {
+            presentationDefinition ->
+            // Limit to only this response' presentationDefinition
+            presentationDefinition.id == this.requestedVcPresentationDefinitionId
+        }.map {
+            presentationDefinition ->
+            // Filter to only matching inputDescriptor(s)
+            presentationDefinition.credentialPresentationInputDescriptors
+                .filter {
+                    inputDescriptor ->
+                    inputDescriptor.id == verifiedIdRequirement.id
+                }
+        }.reduce {
+            allInputDescriptors, inputDescriptor ->
+            // Flatten the inputDescriptors into a single array
+            allInputDescriptors + inputDescriptor
+        }
+    if (matchingInputDescriptors.isEmpty())
         throw IdInVerifiedIdRequirementDoesNotMatchRequestException("Id in VerifiedId Requirement does not match the id in request.")
-    if (credentialPresentationInputDescriptor.size > 1)
+    if (matchingInputDescriptors.size > 1)
         throw VerifiedIdRequirementIdConflictException("Multiple VerifiedId Requirements have the same Ids.")
     verifiedIdRequirement.validate().getOrThrow()
-    requestedVcPresentationSubmissionMap[credentialPresentationInputDescriptor.first()] =
+    requestedVcPresentationSubmissionMap[matchingInputDescriptors.first()] =
         (verifiedIdRequirement.verifiedId as VerifiableCredential).raw
 }
 
 private fun PresentationResponse.addGroupRequirement(groupRequirement: GroupRequirement) {
     groupRequirement.validate().getOrThrow()
     val requirements = groupRequirement.requirements
     for (requirement in requirements) {
-        addRequirements(requirement)
+        try {
+            addRequirements(requirement)
+        } catch (exception: IdInVerifiedIdRequirementDoesNotMatchRequestException) {
+            // This will be thrown if a verified ID couldn't be matched which may happen
+            // with multiple VPs
```

Comment:
Could you add a debug log statement here so that we know if we need for any debugging?

Reply:
After further thoughts and discussions, this entire approach has been removed as it should have been handled by knowing the outer-most Group requirement was placed because of the multiple-VP format.